src/ui/EditableTask.ts

```diff
@@ -34,30 +37,30 @@ export class EditableTask {
     blocking: Task[];
 
     private constructor(editableTask: {
-        addGlobalFilterOnSave: boolean;
-        originalBlocking: Task[];
-
-        // NEW_TASK_FIELD_EDIT_REQUIRED
+        dueDate: string;
         description: string;
-        status: Status;
-        priority: EditableTaskPriority;
+        scheduledDate: string;
         recurrenceRule: string;
+        blockedBy: Task[];
+        priority: 'none' | 'lowest' | 'medium' | 'highest' | 'high' | 'low';
+        // onCompletion: 'ignore' | 'delete';
+        onCompletion: OnCompletion;
+        addGlobalFilterOnSave: boolean;
+        originalBlocking: Task[];
         createdDate: string;
-        startDate: string;
-        scheduledDate: string;
-        dueDate: string;
+        blocking: Task[];
+        forwardOnly: boolean;
         doneDate: string;
+        startDate: string;
         cancelledDate: string;
-        forwardOnly: boolean;
-        blockedBy: Task[];
-        blocking: Task[];
+        status: Status;
```

Comment:
Hi Tom,

I know it's a long time ago now, but I don't suppose you kept any notes about what you did in this commit and why?

I'm asking as I've spent about an hour trying to understand this particular change.

I started looking at after trying to merge the latest `main` in to this branch, and found that the reordering of code here made that merge difficult.

I've tried understanding this diff  by doing things like:
- repeating your previous merge from main, in case that broke anything
- re-ordering the lines here to try and see what changed and why

Other than the reordering, I can undestand most of the differences, but I am very nervous about the diff from:

```
priority: EditableTaskPriority;
```

to:

```
priority: 'none' | 'lowest' | 'medium' | 'highest' | 'high' | 'low';
```

Why was that change made? It's discarding a change that was merged in from `main` but I'm sure you would have done it for a good reason - I would just like to understand why...

Thanks in advance.

Reply:
Don't worry if you can't remember, I'll be able to edit the file to make later merges easier - it's just that if you have any info, it would be appreciated!

Reply:
Ah, I see now that `EditableTaskPriority` is simply this, and not a class.

```ts
type EditableTaskPriority = 'none' | 'lowest' | 'low' | 'medium' | 'high' | 'highest';
```

That means that the change to replace it with the individual types didn't break any behaviour, it just made the code a littler harder to maintain in future...


So I will indeed go ahead and adjust the latest code to revert the revert the reordering of lines in the commit, to make the eventual merge from latest `main` much easier.

So, not need to spend any time on the questions above!

Reply:
Resolving the conversation so I can merge the PR! 🎉